kong/core/router.lua

```diff
@@ -464,7 +464,7 @@ function _M.new(apis)
     method = upper(method)
 
 
-    local host = headers["host"] or headers["Host"]
+    local host = headers["X-Host-Override"] or headers["x-host-override"] or headers["host"] or headers["Host"]
```

Comment:
Can you help me understand how that may affect this case? From what I can read, the `headers` table is only populated by `ngx.req.get_headers`, not passed by reference from anywhere else. So there is no case here where headers are mocked from tests, and in this condition `headers` will _only_ ever be a metatable with the lowercase/underscore behavior.